components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1764,6 +1764,19 @@ _verify_vars() {
   _verify_vars=true
 }
 
+# **Internal** Verifies that hook files do not contain invalid (CR+LF) line endings
+# and fails the build if any are found.
+_verify_hook_line_endings() {
+  find "$PLAN_CONTEXT/hooks" -type f | while read -r file; do
+    if file "$file" | grep -q CRLF; then
```

Comment:
@jeremymv2 Instead of relying on the human-readable output of `file`, what do you think about using `grep` to find the carriage return character directly?
``` bash
$ ls
bar  baz  foo
$ file foo
foo: ASCII text, with CRLF line terminators
$ file bar
bar: ASCII text, with CRLF line terminators
$ file baz
baz: ASCII text
$ grep --files-with-matches $'\r' ./*
./bar
./foo
```
The nice thing about that is that you could find _all_ offending files and report back to the user, rather than failing at the first one.

Thanks for tackling this, by the way... little things like this can be so hard to debug, and having Habitat catch this for users is a huge win!